components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/EnvironmentMappingUtil.java

```diff
@@ -50,8 +61,44 @@ public static EnvironmentDTO fromEnvironmentToDTO(Environment environment) {
         environmentDTO.setType(environment.getType());
         environmentDTO.setServerUrl(environment.getServerURL());
         environmentDTO.setShowInApiConsole(environment.isShowInConsole());
+        environmentDTO.setProvider(environment.getProvider());
         environmentDTO.setVhosts(environment.getVhosts().stream().map(EnvironmentMappingUtil::fromVHostToVHostDTO)
                 .collect(Collectors.toList()));
+        environmentDTO.setAdditionalProperties(fromAdditionalPropertiesToAdditionalPropertiesDTO
+                (environment.getAdditionalProperties()));
+
+        if (SolaceConstants.SOLACE_ENVIRONMENT.equalsIgnoreCase(environment.getProvider())) {
```

Comment:
do we need hardcode solace

Reply:
To show the available endpoint URLs on the publisher portal for Solace deployments we need to use the Solace Admin APIs. Hence this part is needed.